var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -139,21 +141,14 @@ def install(self, spec, prefix):
                 '--disable-hl',
             ])
 
-        configure(
-            "--prefix=%s" % prefix,
-            "--with-zlib=%s" % spec['zlib'].prefix,
-            *extra_args)
-        make()
+        return ["--with-zlib=%s" % spec['zlib'].prefix] + extra_args
 
-        if self.run_tests:
-            make("check")
-
-        make("install")
-        self.check_install(spec)
-
-    def check_install(self, spec):
+    @AutotoolsPackage.sanity_check('install')
+    @AutotoolsPackage.on_package_attributes(run_tests=True)
+    def check_install(self):
```

Comment:
Apparently I cannot reply above for some reason, so concerning `configure` & Co. : 

 1. The signature is like that to preserve backward compatibility, as `install` had the same signature
 1. In the meanwhile I checked, and I couldn't find cases where `spec != self.spec` and `prefix != self.prefix`
 1. I really hate useless arguments, did I mention it ? :smile: 

As we are instructing users how to package things via `spack create` and `documentation` I would rather add a `spec = self.spec` line within the body of `configure_args` in the `AutotoolsPackage` template rather than adding a redundant argument to the function call.